gpu-aware-scheduling/pkg/gpuscheduler/scheduler.go

```diff
@@ -403,38 +402,45 @@ func (m *GASExtender) bindNode(args *extender.BindingArgs) *extender.BindingResu
 	annotation, err := m.runSchedulingLogic(pod, args.Node)
 	resourcesAdjusted := false
 
-	if err == nil {
-		klog.V(L3).Infof("bind %v:%v to node %v annotation %v", args.PodNamespace, args.PodName, args.Node, annotation)
-		// adjust (add) resources in cache, this includes sanity check for the inputs
-		err = m.cache.adjustPodResourcesL(pod, add, annotation, args.Node)
+	defer func() { // deferred errorhandler
+		if err != nil {
+			klog.Error("binding failed:", err.Error())
+			result.Error = err.Error()
 
-		if err == nil {
-			resourcesAdjusted = true
-			// annotate POD with per-container GPU selection
-			err = m.annotatePodBind(annotation, pod)
-
-			if err == nil {
-				// now bind
-				binding := &v1.Binding{
-					ObjectMeta: metav1.ObjectMeta{Name: args.PodName, UID: args.PodUID},
-					Target:     v1.ObjectReference{Kind: "Node", Name: args.Node},
-				}
-				opts := metav1.CreateOptions{}
-				err = m.clientset.CoreV1().Pods(args.PodNamespace).Bind(context.TODO(), binding, opts)
+			if resourcesAdjusted {
+				// Restore resources to cache. Removing resources should not fail if adding was ok.
+				_ = m.cache.adjustPodResourcesL(pod, remove, annotation, args.Node)
 			}
 		}
+	}()
+
+	if err != nil {
```

Comment:
Maybe move this error up above the deferred func? It's just hanging out here. Or move the deferred function to sit on L400, beside defer.rwmutex.Unlock()